dataactvalidator/validation_handlers/file_generation_manager.py

```diff
@@ -27,129 +20,92 @@ class FileGenerationManager:
+
+    def generate_file(self):
+        """ Generates a file based on the FileGeneration object and updates any Jobs referencing it """
+        raw_filename = CONFIG_BROKER["".join([FILE_TYPE_DICT_LETTER_NAME[self.file_type], "_file_name"])]
+        file_name = S3Handler.get_timestamped_filename(raw_filename)
+        if self.is_local:
+            file_path = "".join([CONFIG_BROKER['broker_files'], file_name])
+        else:
+            file_path = "".join(["None/", file_name])
+
+        # Generate the file and upload to S3
+        if self.file_generation:
+            self.generate_d_file(file_path)
+        elif self.file_type == 'E':
+            self.generate_e_file()
+        else:
+            self.generate_f_file()
+
+        log_data = {
+            'message': 'Finished file {} generation'.format(self.file_type), 'message_type': 'ValidatorInfo',
+            'file_type': self.file_type, 'file_path': file_path
+        }
+        if self.file_generation:
+            log_data.update({
+                'agency_code': self.file_generation.agency_code, 'agency_type': self.file_generation.agency_type,
+                'start_date': self.file_generation.start_date, 'end_date': self.file_generation.end_date,
+                'file_generation_id': self.file_generation.file_generation_id
+            })
+        else:
+            log_data['job_id'] = self.job.job_id
         logger.info(log_data)
 
-        # Get or create a FileRequest for this generation
-        current_date = datetime.now().date()
-        file_request_params = {
-            "job_id": self.job.job_id, "is_cached_file": True, "start_date": self.job.start_date,
-            "end_date": self.job.end_date, "agency_code": self.agency_code, "file_type": self.job.file_type.letter_name,
-            "agency_type": self.agency_type
+    def generate_d_file(self, file_path):
+        """ Write file D1 or D2 to an appropriate CSV. """
+        log_data = {
+            'message': 'Starting file {} generation'.format(self.file_type), 'message_type': 'ValidatorInfo',
+            'agency_code': self.file_generation.agency_code, 'agency_type': self.file_generation.agency_type,
+            'start_date': self.file_generation.start_date, 'end_date': self.file_generation.end_date,
+            'file_generation_id': self.file_generation.file_generation_id, 'file_type': self.file_type,
+            'file_path': file_path
         }
+        logger.info(log_data)
 
-        file_request = self.sess.query(FileRequest).filter_by(**file_request_params).one_or_none()
-        if not file_request:
-            file_request_params["request_date"] = current_date
-            file_request = FileRequest(**file_request_params)
-            self.sess.add(file_request)
-            self.sess.commit()
-
-        # Mark this Job as not from-cache, and mark the FileRequest as the cached version (requested today)
-        self.job.from_cached = False
-        file_request.is_cached_file = True
```

Comment:
Did this logic persist in migrating to `FileGeneration`. i.e. where a new file gen "requested today" defaults to having `is_cache_file = True`?

Reply:
https://github.com/fedspendingtransparency/data-act-broker-backend/blob/51f019ea7b35e15b2495734651846d06d5db10c1/dataactbroker/helpers/generation_helper.py#L67